cmake/CompileRocksDB.cmake

```diff
@@ -4,13 +4,26 @@ find_package(RocksDB 6.27.3)
 
 include(ExternalProject)
 
+if (USE_LIBCXX AND NOT APPLE)
+  set(ROCKS_CMAKE_CXX_FLAGS "-stdlib=libc++")
+else()
+  set(ROCKS_CMAKE_CXX_FLAGS "")
+endif()
+
+# Note:  If you find yourself reading this because you can't link to
```

Comment:
I tried forwarding CMAKE_CXX_FLAGS, CMAKE_STATIC_LINKER_FLAGS and CMAKE_SHARED_LINKER_FLAGS, they will not fix the issue that linking would fail.

Reply:
I also hit all sorts of weird issues when I tried passing through the entire CMAKE_CXX_FLAGS variable, and when I didn't delete the environment variables I listed above.